src/hooks/useLocationState.ts

```diff
@@ -0,0 +1,18 @@
+import { appsSection } from "@saleor/apps/urls";
+import { matchPath, useLocation } from "react-router";
+
+const isAppPath = (pathname: string) =>
+  !!matchPath(pathname, {
+    path: `${appsSection}:id`,
+  });
+
+/*
+ * Use detailed information about the current location.
+ */
+export const useLocationState = () => {
+  const location = useLocation();
+
+  return {
+    isAppPath: isAppPath(location.pathname),
+  };
```

Comment:
I don't see the generic uses of this hook outside of apps, I'd put it in `src/apps` or `src/apps/utils`

Reply:
Agreed